syntax = "proto3";
import "google/protobuf/wrappers.proto";
import "google/protobuf/empty.proto";

// 定义命名空间，一般是项目名或者解决方案名
option csharp_namespace = "HandDay.Grpc.Initialization";

service Initialization
{
    //脚本初始化
    rpc InitSqlScript(G_WhiteListCollectonRequest) returns(G_CheckErpConnectionResponse); 

    //检查 ERP是否在线，判断数据库是否连接成功
    rpc CheckErpConnection(google.protobuf.Empty) returns (G_CheckErpConnectionResponse);
    
    //获取数据库列表信息
    rpc GetDataBaseList(google.protobuf.Empty) returns(G_DataBaseListResponse);

    //获取客户白名单列表
    rpc GetCustomerWhiteList(google.protobuf.Empty) returns (google.protobuf.StringValue);

    //增加白名单配置
    rpc SetCustomerWhiteList(google.protobuf.StringValue) returns (google.protobuf.StringValue);

    //移出白名单配置
    rpc RemoveCustomerWhiteList(google.protobuf.StringValue) returns (google.protobuf.StringValue);

    //GRPC性能测试接口
    rpc TestGRPCPorwer(google.protobuf.Empty) returns (google.protobuf.StringValue); 

    //GRPC 版本
    rpc GetGrpcVerion(google.protobuf.Empty) returns (google.protobuf.StringValue);
}

//ERP白名单信息
message G_WhiteListRequest
{
    double CorpId=1;//CorpId
    double ErpBId=2;//暂时用来做记录用
    string Key=3;//白名单项目Key
    int32 Value=4;//白名单项目是否启用
}

//ERP白名单信息列表
message G_WhiteListCollectonRequest
{
    repeated G_WhiteListRequest List=1;
    int64 CorpId=2;//CorpId
}

//ERP是否在线返回参数
message G_CheckErpConnectionResponse
{
    bool Success =1;
    string Msg=2;
}

//获取数据库列表信息 c
message G_DataBaseListRequest {
    string JxcDBAddress=1;//访问地址
    string JxcDBAccount=2;//访问账号
    string JxcDBPwd=3;//访问密码
}

//数据库列表
message G_DataBaseListResponse
{
    repeated string DataBase=1;
}